doc/api/crypto.md

```diff
@@ -1926,8 +1929,8 @@ changes:
   - `publicKey`: {string | Buffer | KeyObject}
   - `privateKey`: {string | Buffer | KeyObject}
 
-Generates a new asymmetric key pair of the given `type`. Only RSA, DSA and EC
-are currently supported.
+Generates a new asymmetric key pair of the given `type`. RSA, DSA, EC, Ed25519
+and Ed448 are currently supported.
```

Comment:
The two new types are missing from the `type` option above.

I'm still strugglying to understand why we wouldn't just do `generateKeyPair('ec', {namedCurve: 'Ed448'}, ...`. I asked OpenSSL why they made a new key type.

Reply:
I had the same question. That being said atm i wouldn’t know if the key is ES521 or Ed25519 due to lack of details about the keyobject instance.

Reply:
Or `sectpXXX`... I'd be happy with a PR to add `.fields` to key objects :-) if you have the energy.

Reply:
I have the energy and motivation but I failed to understand or navigate the openssl documentation to find the methods to use in order to get the key components. Any tips and pointers are welcome of course.

Reply:
Read X509ToObject in node_crypto.cc. I didn't even try to look at openssl API docs for the EC parts of that function, I found an `openssl` CLI command that printed the info I wanted, then read its source.

Reply:
> I'm still strugglying to understand why we wouldn't just do `generateKeyPair('ec', {namedCurve: 'Ed448'}, ...`.

I don't know why OpenSSL did it, but to me, it makes sense, since their ASN.1 encoding is different and EdDSA curves have a different OID.

> I'd be happy with a PR to add .fields to key objects

I'd love to work on that! (If that's okay for @panva.)

Reply:
It makes sense. But i’d really need some pointers to what curves the keys are

Reply:
Sure thing @tniessen.